crates/bcr-wdc-ebpp/src/ebill.rs

```diff
@@ -1,25 +1,64 @@
 // ----- standard library imports
 // ----- extra library imports
 use async_trait::async_trait;
-use bdk_wallet::bitcoin::Amount;
+use bcr_wdc_ebill_client::{EbillClient, Url};
+use bdk_wallet::{
+    bitcoin::{Amount, PrivateKey},
+    keys::KeyMap,
+    miniscript::descriptor,
+};
 // ----- local imports
-use crate::error::Result;
+use crate::error::{Error, Result};
 use crate::service::EBillNode;
 
 // ----- end imports
 
-pub struct DummyEbillNode;
+#[derive(Debug, Clone, serde::Deserialize)]
+pub struct EBillClientConfig {
+    base_url: Url,
+}
+
+#[derive(Debug, Clone)]
+pub struct EBillClient(EbillClient);
+impl EBillClient {
+    pub fn new(cfg: EBillClientConfig) -> Self {
+        let cl = EbillClient::new(cfg.base_url);
+        Self(cl)
+    }
+}
 
 #[async_trait]
-impl EBillNode for DummyEbillNode {
+impl EBillNode for EBillClient {
     async fn request_to_pay(&self, bill: &str, amount: Amount) -> Result<String> {
         tracing::info!(
-            "DummyEbillNode: request_to_pay called with bill: {}, amount: {}",
+            "EBillClient: request_to_pay called with bill: {}, amount: {}",
             bill,
             amount
         );
-        Ok(String::from(
-            "wpkh(Ky1BY5QkB6xb3iQQjJQmVcvqc6mkLBaZTW1xCWpf91aFGBh1kyQ7)",
-        ))
+        let request = bcr_wdc_webapi::bill::RequestToPayBitcreditBillPayload {
+            bill_id: String::from(bill),
+            currency: String::from("BTC"),
```

Comment:
Currently, we validate currency to only accept "sat" as valid - this will change in the future, though and then it'll likely be an enum